### ISSUE TITLE
Build failure: opensplat

### ISSUE THREAD

**Issue by Rijndael1998** [NixOS/nixpkgs#414981]
### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

run `nix-shell -j 1 -p opensplat`

or include `opensplat` in your `configuration.nix`'s packages.

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/job/nixpkgs/staging-next/opensplat.x86_64-linux

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/n5n7gm23qk100wnl9bh8xklbyrz4kwfw-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/z7n6i4qzxchwdwkv1d6akkbbiggn9ryi-opensplat-1.1.4/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/z7n6i4qzxchwdwkv1d6akkbbiggn9ryi-opensplat-1.1.4/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/z7n6i4qzxchwdwkv1d6akkbbiggn9ryi-opensplat-1.1.4/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/z7n6i4qzxchwdwkv1d6akkbbiggn9ryi-opensplat-1.1.4/share/doc/opensplat -DCMAKE_INSTALL_INFODIR=/nix/store/z7n6i4qzxchwdwkv1d6akkbbiggn9ryi-opensplat-1.1.4/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/z7n6i4qzxchwdwkv1d6akkbbiggn9ryi-opensplat-1.1.4/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/z7n6i4qzxchwdwkv1d6akkbbiggn9ryi-opensplat-1.1.4/include -DCMAKE_INSTALL_SBINDIR=/nix/store/z7n6i4qzxchwdwkv1d6akkbbiggn9ryi-opensplat-1.1.4/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/z7n6i4qzxchwdwkv1d6akkbbiggn9ryi-opensplat-1.1.4/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/z7n6i4qzxchwdwkv1d6akkbbiggn9ryi-opensplat-1.1.4/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/0fsnicvfpf55nkza12cjnad0w84d6ba7-gcc-wrapper-14.2.1.20250322/bin/strip -DCMAKE_RANLIB=/nix/store/0fsnicvfpf55nkza12cjnad0w84d6ba7-gcc-wrapper-14.2.1.20250322/bin/ranlib -DCMAKE_AR=/nix/store/0fsnicvfpf55nkza12cjnad0w84d6ba7-gcc-wrapper-14.2.1.20250322/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/z7n6i4qzxchwdwkv1d6akkbbiggn9ryi-opensplat-1.1.4 -DCMAKE_SKIP_RPATH:BOOL=TRUE -DFETCHCONTENT_TRY_FIND_PACKAGE_MODE:STRING=ALWAYS -DPROTOC_EXE=/nix/store/88ma4lbybcpdg0z8745nw9mvj5anb7mq-protobuf-30.2/bin/protoc -DProtobuf_PROTOC_EXE=/nix/store/88ma4lbybcpdg0z8745nw9mvj5anb7mq-protobuf-30.2/bin/protoc -DProtobuf_PROTOC_EXECUTABLE=/nix/store/88ma4lbybcpdg0z8745nw9mvj5anb7mq-protobuf-30.2/bin/protoc -DPYBIND11_PYTHONLIBS_OVERWRITE=OFF -DPYTHON_EXECUTABLE=/nix/store/8w718rm43x7z73xhw9d6vh8s4snrq67h-python3-3.12.10/bin/python3.12 -DPYTHON_INCLUDE_DIR=/nix/store/8w718rm43x7z73xhw9d6vh8s4snrq67h-python3-3.12.10/include/python3.12 -DPYTHON_SITE_PACKAGES=/nix/store/8w718rm43x7z73xhw9d6vh8s4snrq67h-python3-3.12.10/lib/python3.12/site-packages
-- The C compiler identification is GNU 14.2.1
-- The CXX compiler identification is GNU 14.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/0fsnicvfpf55nkza12cjnad0w84d6ba7-gcc-wrapper-14.2.1.20250322/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/0fsnicvfpf55nkza12cjnad0w84d6ba7-gcc-wrapper-14.2.1.20250322/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
OpenSplat Version: 1.1.4
-- Could not find nvcc, please set CUDAToolkit_ROOT.
CMake Warning at CMakeLists.txt:68 (message):
  CUDA toolkit not found, building with CPU support only


CMake Warning at /nix/store/g6raadfrw1ggpn6adab5yb16rvv0cnhw-python3.12-torch-2.7.0-dev/share/cmake/Torch/TorchConfig.cmake:22 (message):
  static library kineto_LIBRARY-NOTFOUND not found.
Call Stack (most recent call first):
  /nix/store/g6raadfrw1ggpn6adab5yb16rvv0cnhw-python3.12-torch-2.7.0-dev/share/cmake/Torch/TorchConfig.cmake:125 (append_torchlib_if_found)
  CMakeLists.txt:164 (find_package)


-- Found Torch: /nix/store/y4jd65pvgmbnfl81xp9slh8r82yn5vhn-python3.12-torch-2.7.0-lib/lib/libtorch.so
-- Found OpenCV: / (found version "4.11.0")
-- Configuring done (0.7s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025
    PROTOC_EXE
    PYBIND11_PYTHONLIBS_OVERWRITE
    PYTHON_EXECUTABLE
    PYTHON_INCLUDE_DIR
    PYTHON_SITE_PACKAGES
    Protobuf_PROTOC_EXE
    Protobuf_PROTOC_EXECUTABLE


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j16
[1/19] Building CXX object CMakeFiles/opensplat.dir/utils.cpp.o
[2/19] Building CXX object CMakeFiles/opensplat.dir/project_gaussians.cpp.o
[3/19] Building CXX object CMakeFiles/opensplat.dir/optim_scheduler.cpp.o
[4/19] Building CXX object CMakeFiles/opensplat.dir/tensor_math.cpp.o
FAILED: CMakeFiles/opensplat.dir/tensor_math.cpp.o 
/nix/store/0fsnicvfpf55nkza12cjnad0w84d6ba7-gcc-wrapper-14.2.1.20250322/bin/g++ -DCXXOPTS_USE_UNICODE -DUSE_C10D_GLOO -DUSE_DISTRIBUTED -DUSE_RPC -DUSE_TENSORPIPE -I/build/source/rasterizer -isystem /nix/store/g6raadfrw1ggpn6adab5yb16rvv0cnhw-python3.12-torch-2.7.0-dev/include/torch/csrc/api/include -isystem /nix/store/9wd1ckcv6ambal1qka0jlndxg0px1660-opencv-4.11.0/include/opencv4 -O3 -DNDEBUG -std=gnu++17 -DAPP_VERSION=\"1.1.4\" -DAPP_REVISION=\"dev\" -D_GLIBCXX_USE_CXX11_ABI=1 -I/nix/store/fs735n061q3ikx81yx7fpksd330gdcn0-icu4c-74.2-dev/include -MD -MT CMakeFiles/opensplat.dir/tensor_math.cpp.o -MF CMakeFiles/opensplat.dir/tensor_math.cpp.o.d -o CMakeFiles/opensplat.dir/tensor_math.cpp.o -c /build/source/tensor_math.cpp
/build/source/tensor_math.cpp: In function 'at::Tensor rodriguesToRotation(const at::Tensor&)':
/build/source/tensor_math.cpp:71:26: error: 'torch::linalg' has not been declared
   71 |     float theta = torch::linalg::vector_norm(rodrigues, 2, { -1 }, true, torch::kFloat32).item<float>();
      |                          ^~~~~~
/build/source/tensor_math.cpp:71:96: error: expected primary-expression before 'float'
   71 |     float theta = torch::linalg::vector_norm(rodrigues, 2, { -1 }, true, torch::kFloat32).item<float>();
      |                                                                                                ^~~~~
[5/19] Building CXX object CMakeFiles/opensplat.dir/model.cpp.o
FAILED: CMakeFiles/opensplat.dir/model.cpp.o 
/nix/store/0fsnicvfpf55nkza12cjnad0w84d6ba7-gcc-wrapper-14.2.1.20250322/bin/g++ -DCXXOPTS_USE_UNICODE -DUSE_C10D_GLOO -DUSE_DISTRIBUTED -DUSE_RPC -DUSE_TENSORPIPE -I/build/source/rasterizer -isystem /nix/store/g6raadfrw1ggpn6adab5yb16rvv0cnhw-python3.12-torch-2.7.0-dev/include/torch/csrc/api/include -isystem /nix/store/9wd1ckcv6ambal1qka0jlndxg0px1660-opencv-4.11.0/include/opencv4 -O3 -DNDEBUG -std=gnu++17 -DAPP_VERSION=\"1.1.4\" -DAPP_REVISION=\"dev\" -D_GLIBCXX_USE_CXX11_ABI=1 -I/nix/store/fs735n061q3ikx81yx7fpksd330gdcn0-icu4c-74.2-dev/include -MD -MT CMakeFiles/opensplat.dir/model.cpp.o -MF CMakeFiles/opensplat.dir/model.cpp.o.d -o CMakeFiles/opensplat.dir/model.cpp.o -c /build/source/model.cpp
/build/source/model.cpp: In member function 'void Model::afterTrain(int)':
/build/source/model.cpp:287:38: error: 'torch::linalg' has not been declared
  287 |         torch::Tensor grads = torch::linalg::vector_norm(xys.grad().detach(), 2, { -1 }, false, torch::kFloat32);
      |                                      ^~~~~~
/build/source/model.cpp:329:63: error: 'torch::linalg' has not been declared
  329 |             torch::Tensor qs = quats.index({splits}) / torch::linalg::vector_norm(quats.index({splits}), 2, { -1 }, true, torch::kFloat32);
      |                                                               ^~~~~~
[6/19] Building CXX object CMakeFiles/opensplat.dir/spherical_harmonics.cpp.o
[7/19] Building CXX object CMakeFiles/opensplat.dir/opensfm.cpp.o
FAILED: CMakeFiles/opensplat.dir/opensfm.cpp.o 
/nix/store/0fsnicvfpf55nkza12cjnad0w84d6ba7-gcc-wrapper-14.2.1.20250322/bin/g++ -DCXXOPTS_USE_UNICODE -DUSE_C10D_GLOO -DUSE_DISTRIBUTED -DUSE_RPC -DUSE_TENSORPIPE -I/build/source/rasterizer -isystem /nix/store/g6raadfrw1ggpn6adab5yb16rvv0cnhw-python3.12-torch-2.7.0-dev/include/torch/csrc/api/include -isystem /nix/store/9wd1ckcv6ambal1qka0jlndxg0px1660-opencv-4.11.0/include/opencv4 -O3 -DNDEBUG -std=gnu++17 -DAPP_VERSION=\"1.1.4\" -DAPP_REVISION=\"dev\" -D_GLIBCXX_USE_CXX11_ABI=1 -I/nix/store/fs735n061q3ikx81yx7fpksd330gdcn0-icu4c-74.2-dev/include -MD -MT CMakeFiles/opensplat.dir/opensfm.cpp.o -MF CMakeFiles/opensplat.dir/opensfm.cpp.o.d -o CMakeFiles/opensplat.dir/opensfm.cpp.o -c /build/source/opensfm.cpp
/build/source/opensfm.cpp: In function 'InputData osfm::inputDataFromOpenSfM(const std::string&)':
/build/source/opensfm.cpp:96:37: error: 'torch::linalg' has not been declared
   96 |         unorientedPoses[i] = torch::linalg::inv(w2c);
      |                                     ^~~~~~
[8/19] Building CXX object CMakeFiles/opensplat.dir/cv_utils.cpp.o
[9/19] Building CXX object CMakeFiles/opensplat.dir/rasterize_gaussians.cpp.o
[10/19] Building CXX object CMakeFiles/opensplat.dir/ssim.cpp.o
[11/19] Building CXX object CMakeFiles/opensplat.dir/point_io.cpp.o
[12/19] Building CXX object CMakeFiles/opensplat.dir/kdtree_tensor.cpp.o
[13/19] Building CXX object CMakeFiles/gsplat_cpu.dir/rasterizer/gsplat-cpu/gsplat_cpu.cpp.o
[14/19] Building CXX object CMakeFiles/opensplat.dir/colmap.cpp.o
[15/19] Building CXX object CMakeFiles/opensplat.dir/input_data.cpp.o
[16/19] Building CXX object CMakeFiles/opensplat.dir/nerfstudio.cpp.o
[17/19] Building CXX object CMakeFiles/opensplat.dir/opensplat.cpp.o
ninja: build stopped: subcommand failed.
```

### Additional context

Not sure if I linked the right hydra, but they're all broken. My full config is here: https://github.com/Rijndael1998/NixDot/tree/smol-latest

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.15.1, NixOS, 25.11 (Xantusia), 25.11pre811203.d3d2d80a2191`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - channels(root): `"nixos, nixos-unstable"`
 - nixpkgs: `/nix/store/cqc0v97xrvc6pvy6lxvrk9ly0k3w1x8b-nixos/nixos`

### Notify maintainers

@jcaesar 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by jcaesar** [comment]
#406869

**Comment by Rijndael1998** [comment]
#406869 fixes this
